### DISCUSSION TITLE
Problems uploading the report (commit function)

### DISCUSSION THREAD

**Discussion by ghost** [TheAlbertDev/MASB#43] [Q&A] [answered]
I'm trying to upload the REPORT of Arduino but it returns the following error:

![error](https://user-images.githubusercontent.com/78802420/108620684-0f5cf500-742e-11eb-8308-69d336e97915.png)

As the error's description indicates, I deleted the file "REPORT" and copied it again but still the same error appeared. Afterwards, I realized I was not in the branch A-julia since the blue numbers next to the path correspond to the "RETO STM" identifier, as can be seen in the next picture: 

![git log](https://user-images.githubusercontent.com/78802420/108620735-7b3f5d80-742e-11eb-85e4-72de5e50806c.png)

I've tried to access the A-julia branch with the command I used to create the branch: checkout -b develop/A-julia;  but it's useless since it says that this branch is already existing. When I run "git status", it says that the "HEAD is detached at 03ba27e" (= RETO STM): 

![git status](https://user-images.githubusercontent.com/78802420/108620811-de30f480-742e-11eb-8543-b48418ec6b55.png)

Could you tell me how can I access the A-julia branch? Or, if not, why am I having this problem? 

Thanks.

Júlia Meca.

**Comment by Albert-Alvarez** [comment]
Hi Julia,

To switch to other branch you have to execute the command:

```bash
git checkout <branch-name>
```

In this case, without the `-b` parameter. This parameter is indicated to create a branch and, then, switch to the created branch. As your branch already exists, you must not add the `-b` parameter.

Let me know if you have been able to switch to your branch and if your problems have been solved by doing so.

**Comment by ghost** [comment]
Hello again, 

I tried the command you mentioned. I didn't know for sure if I only had to write "A-julia" or "develop/A-julia" as the name of the branch. So I tried both. 

In both cases, I get the same error as before. Look: 

![checkout branch](https://user-images.githubusercontent.com/78802420/108630021-14886700-7463-11eb-850b-cf4afd8c0158.JPG)

What should I do? Thank you.

**Comment by Albert-Alvarez** [comment]
It seems that, as the information in the terminal indicates, a git process has crashed. Git creates a lock file when it starts a process and deletes it when it finishes. It does this to prevent other git processes from running if one is already running (by checking whether such a lock file exists before starting a process). Seems that one of those processes has crashed before finishing and the lock file has been left without deleting it. Let's remove it. Execute the following command and let us know if this has solved your problem.

```bash
rm -f .git/index.lock
```

**Comment by ghost** [comment]
After trying this command, It still does not recognize the branch A-julia. Moreover, I tried to access the arduino folder I have inside masb-p01-blaugrana; but it says it doesn't exist. 

![rm](https://user-images.githubusercontent.com/78802420/108630781-dc832300-7466-11eb-880a-bcea09546d56.JPG)

To ensure the REPORT files were still there, I ran a "git status" and confirmed they are still there (in fact, it even recognized the folder arduino here)

![status](https://user-images.githubusercontent.com/78802420/108630806-03d9f000-7467-11eb-97be-020edfef67d7.JPG)

**Comment by Albert-Alvarez** [comment]
But your branch is `A-julia` or `develop/A-julia`?

**Comment by ghost** [comment]
When I created the branch, I wrote: 

`git checkout -b develop/A-julia`

So I believe my branch is called A-julia. Nevertheless, I tried both options and none of them worked. It returns the same error: "A-julia did not match any file(s) known to git". But as you can see in the next image, the branch is created. 

![branch](https://user-images.githubusercontent.com/78802420/108631253-32f16100-7469-11eb-9658-83d653ac773c.JPG)

**Comment by Albert-Alvarez** [comment]
Your branch is called `develop/A-julia`, so you have to use the command:

```bash
git checkout develop/A-julia
```

to jump to your branch.

**Comment by ghost** [comment]
Okay, I do so. 

![develop](https://user-images.githubusercontent.com/78802420/108631732-97152480-746b-11eb-9802-752a90bd7150.JPG)

Even then, a new error pops up. How can I fix this new one?

Sorry for the inconvenience.

**Comment by Albert-Alvarez** [comment, marked as answer]
You don't have to apologize! 😉

Git is telling you that you have edited a file in your current commit (03ba27e). In this case, the file `masb-p01.ino`. So, it doesn't allow you to switch to another branch without saving this modifications.

If you want to hold the modifications, the easiest way to solve this (not the most correct...) is:

1. Copy/paste the file `masb-p01.ino` to other location outside the folder of your repository (outside the `masb-p01-blaugrana` folder).

2. Restore your file to its original state with this command:

```bash
git checkout -- arduino/masb-p01/masb-p01.ino
``` 

3. Switch to your branch:

```bash
git checkout develop/A-julia
```

4. Copy/paste the previously copy/pasted file from outside your repository to the folder `arduino/masb-p01/`.

**Comment by ghost** [comment]
Thank you! The error doesn't appear anymore! :)
